mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -132,6 +132,12 @@ makeRockConv2D(ConversionPatternRewriter &rw, Operation *op, Value input,
   std::optional<uint32_t> num_cu;
   rock::GemmFeatures features;
   std::tie(arch, num_cu, features) = getArchAttributes(op);
+  // Even if acceleration is supported by the arch, datatype might not be
+  // valid accelerated data types
+  if (!rock::isAccel(features, input.getType())) {
+    features = bitEnumClear(features, rock::GemmFeatures::mfma |
+                                          rock::GemmFeatures::wmma);
+  }
```

Comment:
Following the last comment, I don't quite like this approach that in middle of lowering we'd override the `features` in unsupported scenarios. If we want to have such kind of conditions it should happen very early in a pipeline (or before the lowering started, like in conv2d generator).